torchrl/envs/common.py

```diff
@@ -334,6 +334,14 @@ def run_type_checks(self) -> bool:
     def run_type_checks(self, run_type_checks: bool) -> None:
         self._run_type_checks = run_type_checks
 
+    @property
+    def allow_done_after_reset(self) -> bool:
+        return self._allow_done_after_reset
+
+    @allow_done_after_reset.setter
+    def allow_done_after_reset(self, allow_done_after_reset: bool):
+        self._allow_done_after_reset = allow_done_after_reset
```

Comment:
why do we want this to be public? We should be super conservative with public attributes because once it's there we can't take it away!

Reply:
So basically if I did not make it public, implementing the getter and setter solution would not be possible.

It is quite convoluted, but let me explain.

imagin i implement private getter and setters for `TransformedEnv` with the setter trhowing the error. Now, when in the constructor we call the `super().__init__()` https://github.com/pytorch/rl/blob/baea10b804f0836c6d95ce0d23e1e0906f49068f/torchrl/envs/transforms/transforms.py#L575
that will try to set `env._allow_done_on_reset` that however now calls the setter of the `TransofrmedEnv`, throwing an error.

Reply:
This is the reason behind my first prroposed solution.

Reply:
A possible solution is that the setter of the transformed env does not trhow an error but sets the variable in the base env. It seems to me that you wanted to avoid this though

Reply:
Why not: set it to `None`, `get` it, if the value is not `None` throw an error, otherwise set the private attribute.
Setting the private attribute will be ok as long as it's `None` (the `setter` of this property will be a no-op that passes silently if the value is `None` and throws an exception otherwise).
The `getter` on the other hand gets it from the env.

Like this we don't change anything, have the error where appropriate and keep the attribute private.

Reply:
What i understand from the second sentence  might work.

i am not super able to understand what you mean in the first sentence.

but i can make the setter a no-op for None values if that is what you are suggesting, that might work. (Maybe just a bit confusing imo)

Reply:
Implemented